owncloudApp/src/main/java/com/owncloud/android/presentation/files/details/FileDetailsFragment.kt

```diff
@@ -448,9 +448,15 @@ class FileDetailsFragment : FileFragment() {
         } else { // Transfer is upload (?)
             getString(R.string.uploader_upload_in_progress_ticker)
         }
+        val workProgress = workInfo.progress.getInt(DownloadFileWorker.WORKER_KEY_PROGRESS, -1)
         binding.fdProgressBar.apply {
-            isIndeterminate = false
-            progress = workInfo.progress.getInt(DownloadFileWorker.WORKER_KEY_PROGRESS, -1)
+            if (workProgress == -1) {
+                isIndeterminate = true
```

Comment:
what do you think about avoiding the global if structure?

binding.fdProgressBar.apply {
        isIndeterminate = (workProgress == -1)
        progress = if (workProgress == -1) progress else workProgress
        invalidate()
    }

Reply:
I just made it like that because progress shouldn't be defined in case `workProgress == -1`, and invalidate as well, those 2 lines should be only executed when `workProgress != -1`. That's why I decided englobe everything in a `if else` structure, to avoid adding the same `if` in every line

Reply:
Right, invalidate is inside a condition :)